pkg/pipeline/stream/source.go

```diff
@@ -177,11 +177,11 @@ func (n *SourceNode) Stop(ctx context.Context, reason error) error {
 		// only execute stop once, more calls won't make a difference
 		err = n.stop(ctx, reason)
 		stopExecuted = true
-		if err != nil {
-			// an error happened, allow stop to be executed again
-			n.stopOnce = sync.Once{}
```

Comment:
is that allowed in any circumstances? replacing the receiver in execution?

Reply:
YOLO:)

Reply:
Don't do this at home, kids!